### PR TITLE
sieve-editor-gui: init at 0.6.1

### PR DESCRIPTION
**Pull request by Silver-Golden**
## Description of changes

A GUI sieve script editor that integrates with Thunderbird.   
Sieve scripts allow for a user controlled automated processing of mail.  

Previously it used to be a plugin for Thunderbird but got split off into its own thing, partly because plugin support waned in Thunderbird as well as to make it better cross platform compatible.

* Initiated at 0.6.1
* homepage: https://github.com/thsmi/sieve
* I have been personally using it for the last few months without issue.
* Used teh nixfmt in the dev shell.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/si/sieve-editor-gui/package.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+      name = pname;
+      exec = pname;
+      desktopName = "Sieve Editor";
+      icon = pname;
+      categories = [ "Utility" ];
+    })
+  ];
+
+  postInstall = ''
+    makeWrapper ${electron}/bin/electron $out/bin/${pname} --add-flags $out/electron/resources/main_esm.js
```
Comment (SuperSandro2000) [inline pkgs/by-name/si/sieve-editor-gui/package.nix:57]:
```suggestion
    makeWrapper ${lib.getExe electron} $out/bin/sieve-editor-gui \
      --add-flags $out/electron/resources/main_esm.js
```

File: pkgs/by-name/si/sieve-editor-gui/package.nix [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+  env = {
+    ELECTRON_SKIP_BINARY_DOWNLOAD = 1;
+  };
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = pname;
+      exec = pname;
+      desktopName = "Sieve Editor";
+      icon = pname;
```
Comment (SuperSandro2000) [inline pkgs/by-name/si/sieve-editor-gui/package.nix:51]:
```suggestion
      name = "sieve-editor-gui";
      exec = "sieve-editor-gui";
      desktopName = "Sieve Editor";
      icon = "sieve-editor-gui";
```

File: pkgs/by-name/si/sieve-editor-gui/package.nix [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+  nativeBuildInputs = [
+    electron
+    copyDesktopItems
+  ];
+
+  dontNpmBuild = true;
+
+  buildPhase = ''
+    ${nodePackages.gulp}/bin/gulp -LLLL app:package
+  '';
```
Comment (SuperSandro2000) [inline pkgs/by-name/si/sieve-editor-gui/package.nix:34]:
```suggestion
  buildPhase = ''
    gulp -LLLL app:package
  '';
```

File: pkgs/by-name/si/sieve-editor-gui/package.nix [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    repo = "sieve";
+    rev = "5879679ed8d16a34af760ee56bfec16a1a322b4e";
+    hash = "sha256-wl6dwKoGan+DrpXk2p1fD/QN/C2qT4h/g3N73gF8sOI=";
+  };
+
+  npmDepsHash = "sha256-a2I9csxFZJekG1uCOHqdRaLLi5v/BLTz4SU+uBd855A=";
+
+  nativeBuildInputs = [
+    electron
+    copyDesktopItems
```
Comment (SuperSandro2000) [inline pkgs/by-name/si/sieve-editor-gui/package.nix:27]:
```suggestion
    copyDesktopItems
    nodePackages.gulp
```

**Comment by Silver-Golden** [comment]
Feck, going to have to put a pin in this for teh moment.  

Was building locally to test it and it kept crashing.   
Thankfully it did (sometimes) get a crash log.
[-20240903-210431.kcrash.txt](https://github.com/user-attachments/files/16854962/-20240903-210431.kcrash.txt)


I've been using it fine over the summer without any hassle, now this.....   
Thanks anyways Sandro

**Comment by Silver-Golden** [comment]
Huh, it works on 12228ff1752d7b7624a54e9c1af4b222b3c1073b but not on a58bc8ad779655e790115244571758e8de055e3d   
Prob something with KDE?

Anyways it builds and works fine with those changes, pushing in a min.

File: pkgs/by-name/si/sieve-editor-gui/package.nix [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+}:
+buildNpmPackage rec {
+  pname = "sieve-editor-gui";
+  version = "0.6.1";
+  nodejs = nodejs_20;
+
+  src = fetchFromGitHub {
+    owner = "thsmi";
+    repo = "sieve";
+    rev = "5879679ed8d16a34af760ee56bfec16a1a322b4e";
```
Comment (Aleksanaa) [inline pkgs/by-name/si/sieve-editor-gui/package.nix:19]:
```suggestion
    rev = version;
```

File: pkgs/by-name/si/sieve-editor-gui/package.nix [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+      categories = [ "Utility" ];
+    })
+  ];
+
+  postInstall = ''
+    makeWrapper ${lib.getExe electron} $out/bin/sieve-editor-gui \
+      --add-flags $out/electron/resources/main_esm.js
+  '';
+
+  meta = with lib; {
```
Comment (Aleksanaa) [inline pkgs/by-name/si/sieve-editor-gui/package.nix:62]:
```suggestion
  meta = {
```

File: pkgs/by-name/si/sieve-editor-gui/package.nix [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    description = "GUI Sieve editor";
+    homepage = "https://github.com/thsmi/sieve";
+    license = licenses.gpl3Only;
```
Comment (Aleksanaa) [inline pkgs/by-name/si/sieve-editor-gui/package.nix:65]:
```suggestion
    license = licenses.agpl3Only;
```

https://github.com/thsmi/sieve?tab=readme-ov-file#license

Comment (Silver-Golden) [reply pkgs/by-name/si/sieve-editor-gui/package.nix]: Not sure how I misread this, thanks for catching.

File: pkgs/by-name/si/sieve-editor-gui/package.nix (hunk 6 above)
Comment (Silver-Golden) [reply pkgs/by-name/si/sieve-editor-gui/package.nix]: Grand, also added ``lib.`` to the lines that required it then,

File: pkgs/by-name/si/sieve-editor-gui/package.nix [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+    })
+  ];
+
+  postInstall = ''
+    makeWrapper ${lib.getExe electron} $out/bin/sieve-editor-gui \
+      --add-flags $out/electron/resources/main_esm.js
+  '';
+
+  meta = with lib; {
+    description = "GUI Sieve editor";
```
[the remark this answers is not in the document]
Comment (Silver-Golden) [reply pkgs/by-name/si/sieve-editor-gui/package.nix]: A far better description, thank ye.

**Review by Aleksanaa — APPROVED** [review]

**Comment by Aleksanaa** [comment]
Failed to build due to hash mismatch. See https://logs.ofborg.org/?key=nixos/nixpkgs.337931&attempt_id=fe49bc7b-36b7-47de-b3e7-9ad995dd8138

```
error: hash mismatch in fixed-output derivation '/nix/store/gr7zxkr15372lcvblaraqwyq8szzc5rp-source.drv':
         specified: sha256-wl6dwKoGan+DrpXk2p1fD/QN/C2qT4h/g3N73gF8sOI=
            got:    sha256-XWvRiGhfw2NeN1wg7AD9pAzEDSj3a7khoIjEa5Fp42Q=
```

**Comment by Silver-Golden** [comment]
Sorry for the delay, got swamped by uni.

@Aleksanaa I am not sure why by that hash (that was committed was the main branch, not the versioned one (from basically 3 years ago)  
It should build fine now without any issues.

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**